idk what 2 do

AleX04Nov opened the pull request.

AleX04Nov committed:
idk what 2 do

AleX04Nov committed:
lez delete everything :c

pinlast commented on config/config.json:
```diff
@@ -1,22 +0,0 @@
-        "15:50",
-        "17:45"
-    ],
-    "greetings": "\u041f\u0440\u0438\u0432\u0456\u0442 *{}*! \u0426\u0435 \u0431\u043e\u0442 \u0434\u043b\u044f \u0448\u0432\u0438\u0434\u043a\u043e\u0433\u043e \u043f\u0435\u0440\u0435\u0433\u043b\u044f\u0434\u0443 \u0440\u043e\u0437\u043a\u043b\u0430\u0434\u0443 \u0433\u0440\u0443\u043f \u0424\u0422\u0406 \u041a\u041f\u0406.\n/help",
```
Конфиг файлы должны быть читаемыми, нельзя никак это в utf-8?

pinlast commented on requirements.txt:
```diff
@@ -1,5 +0,0 @@
-aiogram
```
Нужно указывать конекретные версии, что бы из-за обновы одной из либ все не пошло по пизде
Типо aiogram==2.6.1

pinlast commented on src/db_handler.py:
```diff
@@ -1,268 +0,0 @@
-        self.chat_ids_orm = Base.classes.chat_ids
-
-        self.name_table_orm = Base.classes.name_table
-
```
Не нужно всё разбивать построчно
Все логические блоки нужно держать вместе

pinlast commented on src/db_handler.py:
```diff
@@ -1,268 +0,0 @@
-                break
-            except exc.OperationalError as e:
-                print(e, "\nReconnection")
-                self._session = self.Session()
```
Нельзя просто открывать новые сессии на каждой ошибке. Нужно:
1) Если у тебя IntegrationError сделать rollback
2) Закрыть старую
3) Открывать новую уже при повторном вызове

pinlast commented on src/db_handler.py:
```diff
@@ -1,268 +0,0 @@
-from sqlalchemy import *
```
Глобально комментарий к контроллеру базы
1) Закрывай старые сессии
2) Чекай IntegrityError и делай rollback
3) Одна сесси открываеться и закрываеться в пределах одной функции, лучше не держать глобально сессию в классе, классы могут по разному объявляться и могут начаться ошибки с сессиями

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-        self.timetable        = dict()
-        self.bot              = aiogram.Bot(self.bot_token)
-        self.dp               = aiogram.Dispatcher(self.bot)
-        self.xls              = XlsHandler("../{}/{}".format(
```
Юзай f-string в таких случаях вместо format, функционал у него такой же как у format, но они намного компакнтее

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-        elif not message.from_user.last_name:
-            nameofuser = message.from_user.first_name
-        else:
-            nameofuser  = message.from_user.first_name
```
Вместо того что бы везде юзать message.from_user.first_name, можно просто сдлать переменную first_name и весь блок будет меньше места занимать и не нужно будет так разбивать создание строки

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-        new_group = message.text[8:].strip().lower()
-        new_group = re.sub('[-]', '', new_group)
-        response_message = str()
-        if self.xls.find_timetable(new_group, {}) != {}:
```
Нельзя  `if self.xls.find_timetable(new_group, {}) != {}:`  заменить на `if self.xls.find_timetable(new_group, None)`?

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-            self.db.upd_chat_rozklad(message.chat.id, new_group)
-            response_message = "Група була змінена на {}".format(new_group)
-        else:
-            response_message = "Група не була змінена на {}. Мабуть, її немає в таблиці, чи була допущена помилка, під час її написання.".format(new_group)
```
Разбивай очень длинные строки на несколько строчек кода

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-        response = await self.get_info(message)
-        group = response[0]
-        now = datetime.datetime.now()
-        weeknum = 1 if datetime.date(
```
Попробуй запихнуть  
```
datetime.date(
            now.year,
            now.month,
            now.day
        ).isocalendar()[1] % 2 == self.week_index

```

В какую-то отдельную переменную или что-то такое, потому что сложно очень читать

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-        if group == "NULL":
-            await self.bot.send_message(
-                message.chat.id,
-                "Ваша поточна група: ***NULL***\nСпробуйте додати групу наприкінці запросу. Наприклад: `/nextlesson фі-73` \nАбо задайте групу таким чином: `/rozklad фі-73`",
```
Вынеси в переменную, при ее создании разбей большую строку на несколько строк в коде и потом сюда вставь просто переменную

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-                "ADMIN MODE",
-                parse_mode = 'Markdown'
-            )
-            url     = message.text[message.entities[0].length:].strip()
```
Что за рандомные пробелы перед `=`?
Пройди весь код с flake8

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-            #
-            if request.status_code == 200 and request.headers["Content-Type"] == "application/vnd.ms-excel":
-                with open(
-                    "../{}/{}".format(self.folder, self.filename),
```
f-string

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-            if get_key(
-                self.day_of_week,
-                re.sub(r'[*-: \n ]', "", day_temp)
-            ) == -1:
```
Так плохо
Запихивай результат функции в переменную и уже ее сравнивай с конкретными значениями

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-                    weekday,
-                    weeknum
-                )
-            )[:-1]
```
Очень незаметная штука
Нужно как-то пофиксить

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-
-        try:
-            for i in range(len(self.lesson_end_time)):
-                tdelta = datetime.datetime.strptime(
```
Ой не
Разбивай по переменным

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-
-        now = now.strftime('%X')
-
-        FMT1 = '%H:%M:%S'
```
Это лучше куда-то в settings.py вынести

pinlast replied:
Как и подключение к базе и всякие разные константы

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-        try:
-            while True:
-                for i in range(len(self.lesson_end_time)):
-                    tdelta = datetime.datetime.strptime(
```
По переменным

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -1,820 +0,0 @@
-                add_days += 1
-                if weekday == 7:
-                    weekday = 1
-                    weeknum = 1 if weeknum == 2 else 2
```
Очень сожночитаемо
Нужно как-то отрефакторить

pinlast commented on src/xls_handler.py:
```diff
@@ -1,341 +0,0 @@
-                        res[name_quest]["week: {}".format(i + 1)] = dict()
-                        j = 0
-                        for row in range(4, sheet.nrows):
-                            if (row - 4) % 15 == 0:
```
4 вложеных цыкла и еще 2 if?
Разложи по функциям что бы можно было адекватно с этим работать

pinlast commented on src/xls_handler.py:
```diff
@@ -1,341 +0,0 @@
-                                    break
-                                res[
-                                    name_quest
-                                ][
```
И тогда не нужно будет такое делать

pinlast commented on src/xls_handler.py:
```diff
@@ -1,341 +0,0 @@
-    ):
-        value = table_dict[
-            group
-        ][
```
Не, так не читаемо, нужно что-то другое придумать

pinlast commented on src/xls_handler.py:
```diff
@@ -1,341 +0,0 @@
-                            sheet.cell_value(main_row, main_col + iter)
-                        ) != "":
-                            break
-                        if str(
```
Ой не
По переменным и сравнивай между собой переменные

AleX04Nov committed:
second version of changes

lez see, what do we have...

pinlast commented on src/db_handler.py:
```diff
@@ -0,0 +1,309 @@
+                session().close()
+            except exc.IntegrityError as e:
+                print("update_sql() exc.IntegrityError \t", e)
+                session().rollback()
```
Зочем ты тут новые сессии создаешь каждый раз? Я же говорил про одну сессию на функцию

pinlast replied:
Бля, ну попробуй оставить так как было раньше, с одной сессией, если с сессиями сложно, думаю с твоими нагрузками оно не горит, думаю тут не приниципиально, твой вариант тоже хорош

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -0,0 +1,863 @@
+        group = response[0]
+        weeknum = await self.get_current_week()
+
+        response_message = f"*{group}*\n*Week: {weeknum}*\n" \
```
Лучше засунуть все в `()` и убрать `\` 
```
('long'
'string'
'handling')

```

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -0,0 +1,863 @@
+                self.timetable.clear()
+                self.xls.update(f"../{self.folder}/{self.filename}")
+                response_message = f"<b>Увага!</b>\n" \
+                                   f"Розклад було оновлено.\n" \
```
Если это обычная строка, зачем добавлять `f`?

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -0,0 +1,863 @@
+
+        now = now.strftime('%X')
+
+        FMT1 = '%H:%M:%S'
```
Закинь их куда-то в settings.py, если они у тебя везде повторяються

pinlast commented on src/scheduleBot_Manager.py:
```diff
@@ -0,0 +1,863 @@
+                weekday  += 1
+                add_days += 1
+
+                # если попало на воскресенье, то мы автоматом
```
Комментарии - хорошо, а много комментариев будет еще лучше!
И можно сразу на английском, раз уж на то дело пошло

AleX04Nov committed:
fixed session() mistake

AleX04Nov committed:
ver 2.0

old class 'scheduleBot_Manager' has been splitted on two classes:
'scheduleBot_Manager'
and
'scheduleBot_Manager_TGAPI'
It was made in order to make clearer difference between TG bot functions and functions, that can be used in any other APIs.
'scheduleBot_Manager_TGAPI' is a spiritual successor of an old 'scheduleBot_Manager' class.
